screens/signup/authAirtable.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable prettier/prettier */
+import { nullLiteral } from '@babel/types';
+import BASE from '../../lib/common';
+
+// Takes in Expo-generated token (a string) 
+// and creates a new record in the Push Tokens table
+export const createPushToken = function async(token) {
+  return BASE('Push Tokens').create([
+    {
+      fields: {
+        Token: token
+      }
+    }
+  ]);
+};
+
+// Creates a customer record in Airtable
+export const createCustomer = function async(
+  fName,
+  lName,
+  phoneNumber,
+  password,
+  tokenId
+) {
+  return BASE('Customers').create([
+    {
+      fields: {
+        'First Name': fName,
+        'Last Name': lName,
+        'Phone Number': phoneNumber,
+        Password: password,
+        Points: 0,
+        'Push Tokens': [tokenId]
+      }
+    }
+  ]);
+};
+
+// Used in SignUpScreen. Checks if phone number is already in use.
+export const checkForDuplicateCustomer = function async(phoneNumber) {
+  return new Promise((resolve, reject) => {
+    BASE('Customers')
+      .select({
+        maxRecords: 1,
+        filterByFormula: `SEARCH("${phoneNumber}", {Phone Number})`
+      })
+      .all()
+      .then(records => {
+        if (records.length > 0) {
+          resolve(true);
+        } else {
+          resolve(false);
+        }
+      })
+      .catch(err => reject(err));
+  });
+};
+
+// lookupCustomer searches for users based on their
+// phone numbers in the form (XXX) XXX-XXXX and checks against
+// a correct password. If the user is found, we return their first
+// and last name. Otherwise, we will display an error on the login screen.
+export const lookupCustomer = function async(phoneNumber, password) {
```

Comment:
This function can be written smaller but maybe something to do later so we can keep moving!

Reply:
@kyleqhua , @JohnathanZhou makes a good point -- if you are only obtaining a maxRecord of 1 you can use `.firstPage` instead and check if it's not null and just extract the first element of that record list instead of iterating through the whole list